log.cpp

```diff
@@ -33,6 +33,17 @@ void write_stderr(const String& str, Strings&&... strings) {
 }
 }
 
+#if defined(__sun) && !defined(HAVE_DPRINTF) 
```

Comment:
Rather than adding this here (and the check to configure), I think you can replace the `dprintf` call completely.  There are only two calls and they are down around line 238 of this file:

```
   if (si) {
     dprintf(STDERR_FILENO,
         "Terminating due to signal %d %s "	
         "generated by pid=%d uid=%d. %s (%p)\n",
         signo, w_strsignal(signo), si->si_pid, si->si_uid,
         reason, si->si_value.sival_ptr);
   } else {	
     dprintf(STDERR_FILENO, "Terminating due to signal %d %s. %s\n",	  
       signo, w_strsignal(signo), reason);	    
   }	  
```

Can be replaced with something like:

```
if (si) {
  auto msg = watchman::to<std::string>(
        "Terminating due to signal ", signo, " ", w_strsignal(signo),
         " generated by pid=", si->si_pid, " uid=", si->si_uid,  " ",
         reason, " (", si->si_value.sival_ptr, ")\n");
  ignore_result(write(fd, msg.data(), msg.size()));
} else {
auto msg = watchman::to<std::string>(
        "Terminating due to signal ", signo, " ", w_strsignal(signo),
         " ", reason, "\n")
  ignore_result(write(fd, msg.data(), msg.size()));
}
```